src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_NavigateTo.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Execution;
+using Microsoft.CodeAnalysis.NavigateTo;
+using Microsoft.CodeAnalysis.Remote.Arguments;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    internal partial class CodeAnalysisService
+    {
+        public async Task<SerializableNavigateToSearchResult[]> SearchDocumentAsync(
+             SerializableDocumentId documentId, string searchPattern, byte[] solutionChecksum)
+        {
+            var solution = await RoslynServices.SolutionService.GetSolutionAsync(
+                new Checksum(solutionChecksum), CancellationToken).ConfigureAwait(false);
+
+            var project = solution.GetDocument(documentId.Rehydrate());
```

Comment:
my thinking on the layer was 
" result = await DefaultNavigateToEngineService.SearchDocumentInCurrentProcessAsync(
 +                project, searchPattern, CancellationToken).ConfigureAwait(false);"

this kind of thing being in RemoteWorkspace. ServiceHub project is supposed to only deal with servicehub specific thing and marshaling.